### PR TITLE
Add tab for "Completed" and "On hold" tasks for AMA cases in attorney queues

### PR DESCRIPTION
**Pull request by joeyyang**
Resolves https://github.com/department-of-veterans-affairs/caseflow/issues/7540

### Description
Added tabs for "Assigned," "On hold", and "Completed" to the attorney queue table view.

screenshot 2018-11-07 11 03 28

nov-07-2018 11-03-43

### CODE REVIEW

File: client/COPY.json [hunk 1]
```diff
@@ -73,6 +73,12 @@
   "ATTORNEY_QUEUE_TABLE_SUCCESS_MESSAGE_DETAIL": "If you made a mistake, please email your judge to resolve the issue.",
   "ATTORNEY_QUEUE_TABLE_TASK_NEEDS_ASSIGNMENT_ERROR_MESSAGE": "Please ask your judge to assign this case to you in DAS",
   "ATTORNEY_QUEUE_TABLE_TASK_NO_DOCUMENTS_READER_LINK": "View in Reader",
+  "ATTORNEY_QUEUE_PAGE_ASSIGNED_TAB_TITLE": "Assigned (%d)",
+  "ATTORNEY_QUEUE_PAGE_ON_HOLD_TAB_TITLE": "On Hold (%d)",
+  "ATTORNEY_QUEUE_PAGE_COMPLETE_TAB_TITLE": "Completed",
+  "ATTORNEY_QUEUE_PAGE_ASSIGNED_TASKS_DESCRIPTION": "Cases assigned to you.",
+  "ATTORNEY_QUEUE_PAGE_ON_HOLD_TASKS_DESCRIPTION": "When a hold is complete, cases here will automatically return to the Assigned tab.",
+  "ATTORNEY_QUEUE_PAGE_COMPLETE_TASKS_DESCRIPTION": "Cases with tasks that were marked complete in the last two weeks.",
```
Comment (lowellrex) [inline client/COPY.json]: I think we can reuse the `COLOCATED_QUEUE_PAGE_COMPLETE_TASKS_DESCRIPTION` copy here and rename that bit of copy to something like `QUEUE_PAGE_COMPLETE_TASKS_DESCRIPTION`.

File: client/COPY.json [hunk 2]
```diff
@@ -73,6 +73,12 @@
   "ATTORNEY_QUEUE_TABLE_SUCCESS_MESSAGE_DETAIL": "If you made a mistake, please email your judge to resolve the issue.",
   "ATTORNEY_QUEUE_TABLE_TASK_NEEDS_ASSIGNMENT_ERROR_MESSAGE": "Please ask your judge to assign this case to you in DAS",
   "ATTORNEY_QUEUE_TABLE_TASK_NO_DOCUMENTS_READER_LINK": "View in Reader",
+  "ATTORNEY_QUEUE_PAGE_ASSIGNED_TAB_TITLE": "Assigned (%d)",
+  "ATTORNEY_QUEUE_PAGE_ON_HOLD_TAB_TITLE": "On Hold (%d)",
+  "ATTORNEY_QUEUE_PAGE_COMPLETE_TAB_TITLE": "Completed",
```
Comment (lowellrex) [inline client/COPY.json]: Let's consolidate the text for the attorney queue on hold and completed tab titles to two pieces of copy to something like `QUEUE_PAGE_ON_HOLD_TAB_TITLE` and `QUEUE_PAGE_COMPLETE_TAB_TITLE`. That way if we decide to change one of those values we don't have to change it in a bunch of places.

File: client/app/queue/selectors.js [hunk 3]
```diff
@@ -170,7 +170,11 @@ export const newTasksByAssigneeCssIdSelector = createSelector(
 export const workableTasksByAssigneeCssIdSelector = createSelector(
   [tasksByAssigneeCssIdSelector],
   (tasks: Array<TaskWithAppeal>) => tasks.filter(
-    (task) => task.appeal.isLegacyAppeal || task.status !== TASK_STATUSES.on_hold
+    (task) => {
+      return (task.appeal.isLegacyAppeal ||
+          task.status === TASK_STATUSES.assigned ||
+          task.status === TASK_STATUSES.in_progress);
```
Comment (lowellrex) [inline client/app/queue/selectors.js]: Great change!

File: client/app/queue/AttorneyTaskListView.jsx [hunk 4]
```diff
@@ -99,7 +125,10 @@ const mapStateToProps = (state) => {
   } = state;
 
   return ({
-    tasks: workableTasksByAssigneeCssIdSelector(state),
+    tasks: tasksByAssigneeCssIdSelector(state),
```
Comment (lowellrex) [inline client/app/queue/AttorneyTaskListView.jsx]: I think we can get rid of this `tasks` element here. With these changes we only use `tasks` [on line 59](https://github.com/department-of-veterans-affairs/caseflow/pull/7765/files#diff-d3cc2c9a255961c6635743a9e14b22bbR59), and we can instead use the combination of the other three sets of tasks (workable, on hold, and completed) there to get the same result.

File: client/COPY.json (hunk 2 above)
Comment (joeyyang) [reply client/COPY.json]: Great! Just as a note, this PR only changes the variable names, not the names of the tabs themselves (per our queue terminology meeting today).

**Comment by joeyyang** [comment]
thanks, back to you @lowellrex!

File: client/COPY.json [hunk 5]
```diff
@@ -73,24 +73,25 @@
...
-  "COLOCATED_QUEUE_PAGE_COMPLETE_TAB_TITLE": "Completed",
   "COLOCATED_QUEUE_PAGE_NEW_TASKS_DESCRIPTION": "These are new administrative actions that have been assigned to you.",
   "COLOCATED_QUEUE_PAGE_PENDING_TASKS_DESCRIPTION": "When a hold is complete, the case will appear here. Cases with new documents will also appear here.",
   "COLOCATED_QUEUE_PAGE_ON_HOLD_TASKS_DESCRIPTION": "When a hold is complete, the case will automatically return to the Pending action tab.",
-  "COLOCATED_QUEUE_PAGE_COMPLETE_TASKS_DESCRIPTION": "Cases with tasks that were marked complete in the last two weeks will appear here.",
 
   "ORGANIZATIONAL_QUEUE_PAGE_UNASSIGNED_TAB_TITLE": "Unassigned (%d)",
-  "ORGANIZATIONAL_QUEUE_PAGE_ASSIGNED_TAB_TITLE": "Assigned (%d)",
-  "ORGANIZATIONAL_QUEUE_PAGE_COMPLETE_TAB_TITLE": "Completed",
   "ORGANIZATIONAL_QUEUE_PAGE_UNASSIGNED_TASKS_DESCRIPTION": "Cases owned by the %s team that are unassigned to a person.",
```
Comment (lowellrex) [inline client/COPY.json]: I think we need to update a test that still uses these variables: https://github.com/department-of-veterans-affairs/caseflow/blob/1b9a77a5aab5f1b5847b838e560ff5f1275ae4be/spec/feature/queue/task_queue_spec.rb#L169

Comment (joeyyang) [reply client/COPY.json]: Ah! My find/replace was set to only look at `/client` — fixing now.

File: spec/feature/queue/task_queue_spec.rb [hunk 6]
```diff
@@ -59,7 +70,7 @@
     it "supports custom sorting" do
       docket_number_column_header = page.find(:xpath, "//thead/tr/th[3]/span/span[1]")
       docket_number_column_header.click
-      docket_number_column_vals = page.find_all(:xpath, "//tbody/tr/td[3]/span[3]")
+      docket_number_column_vals = page.find_all(:xpath, "//tbody/tr/td[4]/span[3]")
```
Comment (lowellrex) [inline spec/feature/queue/task_queue_spec.rb]: Do we know why this changed?

**Pull request merged**